### PR TITLE
Add support for BigInt numeric

### PR DESCRIPTION
**Pull request by zcei**
### Description of the Change

TC39 added `BigInt` (currently at stage 3) and it's already available in e.g. Node.
The grammer currently doesn't support it and yields an `invalid.illegal.identifier.js`.

You can see this for example in the [`README` of `tc39/proposal-bigint`](https://github.com/tc39/proposal-bigint)

### Alternate Designs

The octal regex could be expanded to not allow the `n` for the legacy octal definition. As this would have complicated things, I left it out. Let me know whether you think it should be added.

### Benefits

Proper GitHub syntax highlighting with BigInts.

### Possible Drawbacks

Not really applicable as this willt part of the 

### Applicable Issues

n/a

### Other

https://github.com/tree-sitter/tree-sitter-javascript/pull/93

### CODE REVIEW

File: grammars/javascript.cson [hunk 1]
```diff
@@ -1265,7 +1265,7 @@
             (?:\\b[0-9]+(\\.)[0-9]+\\b)|                # 1.1
             (?:\\b[0-9]+(\\.)\\B)|                      # 1.
             (?:\\B(\\.)[0-9]+\\b)|                      # .1
-            (?:\\b[0-9]+\\b(?!\\.))                     # 1
+            (?:\\b[0-9]+n?\\b(?!\\.))                     # 1
```
Comment (50Wliu) [inline grammars/javascript.cson]: :art: keep the comments aligned please

**Comment by zcei** [comment]
@50Wliu can you maybe have another look whenever it fits? 🙏

**Comment by bnoordhuis** [comment]
I don't want to be the bump guy but I just published a module today that uses bigints extensively and it shows up real ugly ([angry red example](https://github.com/bnoordhuis/random-bigint/blob/master/test.js)). It would be great if this got merged. Thanks in advance!

**Comment by bnoordhuis** [comment]
@lee-dohm It's at stage 3 in TC39 and it's shipping in at least one browser (Chrome, flag-gated in Firefox) and Node.js. That's good enough, isn't it? It's out there in the real world, it's not a hypothetical.

(It also seems to have been implemented in Chakracore but I don't know if it's actually been shipped in IE. I'm guessing 'no' what with the transition to a different engine.)

**Comment by lee-dohm** [comment]
> It's at stage 3 in TC39 and it's shipping in at least one browser (Chrome, flag-gated in Firefox) and Node.js. That's good enough, isn't it?

We're aware. We discussed it as a team and decided that we felt that knowing what version of ECMAScript it would appear in would be our bar for inclusion.

**Comment by zcei** [comment]
Hej there 👋 

[TC39 June meeting](https://github.com/tc39/agendas/blob/master/2019/06.md) just happened and BigInt advanced to stage 4 (see the checkmark next to the agenda item) and thus will be included in the next formal spec release (ECMAScript 2020).

The PR towards ecma262 describing BigInt can be found here: https://github.com/tc39/ecma262/pull/1515

I understand you might want to wait until this PR has landed (one never knows what might happen), just wanted to let you know 🙂

**Comment by lkashef** [comment]
Hi @zcei, thank you for your contribution 🙇‍♂️.

However, I am going to close this PR, as it seems that this PR on tree-sitter adds support for BigInt (https://github.com/tree-sitter/tree-sitter-javascript/pull/93).

**Comment by michaelficarra** [comment]
@lkashef The `javascript.cson` changes are still needed for GitHub linguist. Compare [master](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=cson&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fatom%2Flanguage-javascript%2Fmaster%2Fgrammars%2Fjavascript.cson&grammar_text=&code_source=from-text&code_url=&code=let+a+%3D+0n%3B) with [this PR](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=cson&grammar_url=https%3A%2F%2Fraw.githubusercontent.com%2Fzcei%2Flanguage-javascript%2Fgrammar-bigint%2Fgrammars%2Fjavascript.cson&grammar_text=&code_source=from-text&code_url=&code=let+a+%3D+0n%3B).

**Comment by zcei** [comment]
As long as GitHub doesn't (fully?) use tree-sitter, I'd still love to see the changes landing. At least e.g. the example from Ben above still shows up red on GitHub - this visual flaw was my main intention for the contribution.

**Comment by lkashef** [comment]
Thanks @zcei for the context, this issue clearly exist on Github, while I was looking into Atom.

**Comment by molvqingtai** [comment]
Now Optional-Chaining is at Stage 4, TypeScript 3.7 has been officially added, and I hope to add support for him in the future

**Pull request merged**